### PR TITLE
Generate Android versionCode from Major.Minor.Patch

### PR DESCRIPTION
**Pull request by rubenwardy**
Let's never mess up the versionCode again

## To do

This PR is Ready for Review

- [x] Test bump_version.sh
- [x] Do different architectures still need different version codes with app bundles?

### CODE REVIEW

**Comment by rubenwardy** [comment]
Feedback from F-Droid:

> Autoupdates need static values, written somewhere. So that PR will break them, unless we can extract them...

So perhaps bump_version.sh should generate the versionCode?

**Comment by sfan5** [comment]
"Autoupdates" being what? if it just means someone from f-droid has to manually change their metadata then that's not our problem.

**Comment by rubenwardy** [comment]
It appears the watch the repo for `versionCode` changes without configuring gradle. When the versionCode changes, the manifest changes and it automatically tells them about an update. Not sure how much of the release is automated, I think it's just notifications.

So the alternative would be someone having to make an issue every time we release

**Comment by sfan5** [comment]
Surely f-droid can watch tags on our git repo instead?

**Comment by red-001** [comment]
why not just calculate the number as done in this PR and then write it back to the gradle file? Version bumping should only be done with the shell script anyways right?

File: android/build.gradle [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
+project.ext.set("versionMinor", 10)     // Version Minor
 project.ext.set("versionPatch", 0)      // Version Patch
 // ^ keep in sync with cmake
-project.ext.set("versionCode", 48)      // Android Version Code
-// NOTE: +2 after each release!
-// +1 for ARM and +1 for ARM64 APK's, because
-// each APK must have a larger `versionCode` than the previous
+
+project.ext.set("versionBuild", 0)      // Version Build
+// ^ fourth version number to allow releasing Android-only fixes and beta versions
```
Comment (sfan5) [inline android/build.gradle:9]: will that work?
how do we tag these on github? will they be merged into `stable-5`?

**Review by sfan5 — APPROVED** [review]

**Comment by grorp** [comment]
>  Note: The greatest value Google Play allows for versionCode is 2100000000. 
> ~ https://developer.android.com/studio/publish/versioning

so doesn't look like this will cause problems

**Pull request merged**